src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -518,3 +519,31 @@ export const concatAttachmentsAndResponses = (
   response += attachments.reduce((acc, { content }) => acc + content, '')
   return response
 }
+
+// Class to encapsulate maskedData method
+// which returns autoReplyData with masked Corppass UID
+
+export class autoReplyDataObj {
+  data: EmailAutoReplyField[]
+  constructor(autoReplyData: EmailAutoReplyField[]) {
```

Comment:
instead of having a class instance containing the entire array of responses, might it be better to have an array of class instances each holding one response?

in the current implementation, if we have to mask or otherwise modify other fields in future, every class method has to iterate through all the responses. it may be cleaner to just store a single response.

Reply:
see if the revised encapsulation works - not sure if this comment is still relevant. Basically the method maskedAutoReplyData() is called only if we want to retrieve the masked responses for autoreplyfield.